checker/src/org/checkerframework/checker/lock/LockAnnotatedTypeFactory.java

```diff
@@ -511,4 +515,66 @@ protected TreeAnnotator createTreeAnnotator() {
                new ImplicitsTreeAnnotator(this)
         );
     }
+
+    @Override
+    public void annotateImplicit(Element elt, AnnotatedTypeMirror type) {
+        insertLockCheckerGuardedByIfJcipOrJavaxGuardedByIsPresent(elt, type);
+
+        super.annotateImplicit(elt, type);
+    }
+
+    @Override
+    public void annotateImplicit(Tree tree, AnnotatedTypeMirror type, boolean useFlow) {
+        if (tree.getKind() == Tree.Kind.VARIABLE) {
+            insertLockCheckerGuardedByIfJcipOrJavaxGuardedByIsPresent(InternalUtils.symbol((VariableTree) tree), type);
+        }
+
+        super.annotateImplicit(tree, type, useFlow);
+    }
+
+    /**
+     * Given a field declaration with a {@code @net.jcip.annotations.GuardedBy} or
+     * {@code javax.annotation.concurrent.GuardedBy} annotation and an AnnotatedTypeMirror
+     * for that field, inserts the corresponding {@code @org.checkerframework.checker.lock.qual.GuardedBy}
+     * type qualifier into that AnnotatedTypeMirror.
+     *
+     * @param element the Element for a field declaration (this method does nothing if the Element is not for a field declaration)
+     * @param atm the AnnotatedTypeMirror for the field - the {@code @GuardedBy} type qualifier will be inserted here
+     */
+    private void insertLockCheckerGuardedByIfJcipOrJavaxGuardedByIsPresent(Element element, AnnotatedTypeMirror atm) {
```

Comment:
This is a rather lengthy method name.  What about `translateJcipAndJavacAnnotions` instead?  (Though, if you like the longer one better, it's ok)